### PR TITLE
Replace eye with foreman.

### PR DESCRIPTION
**Pull request by skyschub**
Looking at #5966, we've come full circle now. `eye` is effectively dead (least release in 2016, no commits for three years). 

And what's worse, `eye` depends on `kostya-sigar`, which is a `sigar` fork from the author of `eye`, and that's dead, too. `kostya-sigar` fails to build its native dependencies for me on my systems, as [this call to `gettimeofday`](https://github.com/kostya/sigar/blob/9f918d1ae9c32c9037891cdc5cdb0d9800ff93c5/src/sigar_util.c#L980) is made in a file that does not include `sys/time.h`. I'm not sure why building this works on some systems, [but Debian is running into the same build failures](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1071624).

Foreman is a lot simpler than eye, for better or worse. It doesn't require a config file, but it also doesn't allow you to configure pretty much anything. Therefore, this changes the behavior of `script/server` in two severe ways:

1. We no longer redirect log outputs into files. This would be easy to add with a stderr+stdout redirect, but... we get a lot of support requests at the moment where people say "diaspora doesn't start and I don't see any error message". People don't know to check the output files, and so they're confused. I'd say that logging to stdout is actually kinda the default these days - a lot of services do it, and systemd' journaling even kinda expects you to. I'd say this is a feature.
2. Foreman does not actively control services. If, for example, sidekiq crashes, then foreman will send a SIGTERM to puma and then quit. For production pods that run `script/server` in a systemd service with `Restart=always`, that's no big deal, but if someone runs it in a tmux or something, they could be surprised.

Ultimately, I think the only thing that ever used the process management capabilities of `eye` was `diaspora-dev`, which used it to send a `SIGUSR2` to puma for soft restarts. I really don't think we need that complexity.

### CODE REVIEW

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
> It doesn't require a config file

Well, it requires the `Procfile`, but that already exists from the past.

> We no longer redirect log outputs into files.

I agree with your points here, that this probably is an improvement. In addition I like, that I can see which outputs are from the web server and which are from puma. In the dev-env, we already used stdout with eye, but you couldn't directly see if it was from web or puma, so I think this is an additional feature :partying_face: 

> Foreman does not actively control services. ... For production pods that run `script/server` in a systemd service with `Restart=always`, that's no big deal

Do we want to add a more prominent changelog section about this, like we did for the puma change? And do we want to provide an example service file for systemd? Because you also need to make sure that rvm is loaded (as this is still the default installation method in the guides), so you need to wrap the `script/server` in `/bin/bash -lc "..."`.

We also have the systemd examples here, which creates separate services for web (which we also need to update for puma when we release) and sidekiq: https://wiki.diasporafoundation.org/Alternative_startup_methods#systemd Maybe we can just link to there from the changelog, if people want to setup systemd instead of foreman? If you are using systemd, you don't really need to wrap foreman around everything, but the disadvantage is, that you would need to change your services, if we change something (like switching unicorn to puma), while if you just call `script/server` from systemd, we can control everything and podmins don't need to change their services.

File: script/server [hunk 1]
```diff
@@ -162,4 +162,4 @@ fi
 
 # Start Diaspora
 printf "Starting Diaspora in $RAILS_ENV mode with $workers Sidekiq worker(s)."
```
[the remark this answers is not in the document]
Comment (denschub) [reply script/server]: Oh, I didn't even notice that! You have a god ... ... eye. 🥁 

I'm sorry.

[the remark this answers is not in the document]
Comment (SuperTux88) [reply script/server]: To be fair, that was already a problem before, and I never noticed it. :joy:

But now with foreman it was nicely colored and therefore more obvious. :eyes:

**Comment by denschub** [comment]
> Do we want to add a more prominent changelog section about this, like we did for the puma change?

So, not 100% sure. My gut feeling tells me "no", our systemd units in the wiki have been using individual services for quite a long time, and that's how it should be done IMHO. But also, there's no harm in adding a note, so I'm happy to add one. Do you want me to add it directly to the PR, or should we phrase something on IRC and you can add it during merge?

**Review by SuperTux88 — APPROVED** [review]
Looks good now :cookie: 

> Do you want me to add it directly to the PR, or should we phrase something on IRC and you can add it during merge?

As discussed on IRC, I'll add something when merging and link to the wiki, where we now added a simple unit file with `script/server`: https://wiki.diasporafoundation.org/Automatic_startup_methods#Recommended:_systemd

**Pull request merged**